__tests__/genAxios.test.js

```diff
@@ -0,0 +1,125 @@
+const formAxiosMock = require("../__mocks__/gen-axios.mock");
+const responses = [
+  [
+    {
+      type: "get",
+      response: {
+        data: {
+          a: 1
+        },
+        status: 200,
+        statusText: "OK"
+      }
+    },
+    {
+      type: "constructor",
+      response: {
+        data: {
+          b: 2
+        },
+        status: 200,
+        statusText: "OK"
+      }
+    },
+    {
+      type: "post",
+      response: {
+        data: {
+          c: 3
+        },
+        status: 200,
+        statusText: "OK"
+      }
+    }
+  ],
+  [
+    {
+      type: "get",
+      response: {
+        data: {
+          d: 4
+        },
+        status: 200,
+        statusText: "OK"
+      }
+    },
+    {
+      type: "constructor",
+      response: {
+        data: {
+          e: 5
+        },
+        status: 200,
+        statusText: "OK"
+      }
+    },
+    {
+      type: "post",
+      response: {
+        data: {
+          f: 6
+        },
+        status: 200,
+        statusText: "OK"
+      }
+    }
+  ]
+];
+formAxiosMock(responses);
+const { httpSend, httpGET, httpPOST } = require("../adapters/network");
+const axios = require("axios");
+
+const mockMethod1 = async () => {
+  // some random code execution before axios.get
+  const resp1 = await axios.get("http:///wwww.example.com/1");
+  // some random code execution before axios
+  const resp2 = await axios({
+    url: "http://www.example.com/2",
+    method: "post"
+  });
+  // some random code executed before axios.post
+  const resp3 = await axios.post("http://www.example.com/3")
```

Comment:
If I were to interchange these calls, will this test case still pass?

Reply:
No they would not

Reply:
That's a very fragile state to be. We should change it probably.
How about serving the response by matching the request configuration (URL, data, params, headers, etc.)?

Reply:
Don't think it is fragile. When the order of requests is changed, that is like a logical change happening & tests should be able to recognise it IMHO.

Serving response by matching the request configuration is one of the hardest mocking techniques. But they can be easily achieved via using libraries like `supertest`, `nock` etc,.
The limitation I see is that, the generic implementation using such libraries would be a little trickier to implement, as each destination's test-case needs to have it's own implementation of http request mocker

Even in `rudder-server` the mock server is not much different. Please let me know if anything I described is not clear

Reply:
>How about serving the response by matching the request configuration (URL, data, params, headers, etc.)?

We had implemented a mock axios client with a similar intention but I see limitation wrt generalisation

Reply:
The mocked client needn't be generic. I believe we can feed the expected inputs and outputs.

Reply:
No matter what specifics are present in the input, we care more about output that we get. So, I don't think this way we'll have any problem as such.
If this still is not satisfactory, we might need to take the approach taken for mocking axios client in [google_adwords_offline_conversion.test.js](https://github.com/rudderlabs/rudder-transformer/blob/master/__tests__/google_adwords_offline_conversions.test.js#L41-L146)